pkg/apis/workflow/v1alpha1/workflow_phase.go

```diff
@@ -4,17 +4,18 @@ package v1alpha1
 type WorkflowPhase string
 
 const (
-	WorkflowUnknown   WorkflowPhase = ""
-	WorkflowPending   WorkflowPhase = "Pending" // pending some set-up - rarely used
-	WorkflowRunning   WorkflowPhase = "Running" // any node has started; pods might not be running yet, the workflow maybe suspended too
-	WorkflowSucceeded WorkflowPhase = "Succeeded"
-	WorkflowFailed    WorkflowPhase = "Failed" // it maybe that the workflow was terminated
-	WorkflowError     WorkflowPhase = "Error"
+	WorkflowUnknown    WorkflowPhase = ""
+	WorkflowPending    WorkflowPhase = "Pending" // pending some set-up - rarely used
+	WorkflowRunning    WorkflowPhase = "Running" // any node has started; pods might not be running yet, the workflow maybe suspended too
+	WorkflowSucceeded  WorkflowPhase = "Succeeded"
+	WorkflowFailed     WorkflowPhase = "Failed" // it maybe that the workflow was terminated
+	WorkflowError      WorkflowPhase = "Error"
+	WorkflowCompleting WorkflowPhase = "Completing"
 )
 
 func (p WorkflowPhase) Completed() bool {
 	switch p {
-	case WorkflowSucceeded, WorkflowFailed, WorkflowError:
+	case WorkflowSucceeded, WorkflowFailed, WorkflowError, WorkflowCompleting:
```

Comment:
Did you go through all of the calls to `WorkflowPhase.Completed()` (including all of the calls to `WorkflowStatus.Fulfilled()` which calls it) to make sure all of them would be fine with this addition?